src/ServiceManager.php

```diff
@@ -357,7 +357,10 @@ public function configure(array $config)
         }
 
         if (isset($config['invokables']) && ! empty($config['invokables'])) {
-            $this->createAliasesAndFactoriesForInvokables($config['invokables']);
+            $newAliases = $this->createAliasesAndFactoriesForInvokables($config['invokables']);
+            // override existing aliases with those created by invokables to ensure
+            // that they are still present after merging aliases later on
+            $config['aliases'] = $newAliases + ($config['aliases'] ?? []);
```

Comment:
I think `+` should be avoided here, since it has weird merging semantics: https://3v4l.org/mkf2r

Reply:
@Ocramius I don't think so. The only "weird" thing about the `+` operator for arrays is that values from the left operand take precedence over values from the right operand, which is different from the order of parameters in `array_merge`. PHP [documentation](https://www.php.net/manual/en/language.operators.array.php) says this clearly:

>   The + operator returns the right-hand array appended to the left-hand array; for keys that exist in both arrays, the elements from the left-hand array will be used, and the matching elements from the right-hand array will be ignored.

See https://3v4l.org/0gZ0l for an example.

Reply:
Hmm, fairly sure I saw some weird behavior some weeks ago. I'm ok with this scope after this convo, so I'd say :ship: :-)

Overall, `+` for `array` should be avoided in future, but I'll make my case stronger when I got time to investigate further: since it's already in use here, let's merge :+1: